### DISCUSSION TITLE
Ability to exclude matchers from a template

### DISCUSSION THREAD

**Discussion by carlin-q-scott** [projectdiscovery/nuclei-templates#3808] [Ideas]
I'd like to be able to use part of a template by excluding certain matchers. 

For instance, I'm using misconfiguration/http-missing-security-headers.yaml, but I don't want to get results for the less important headers. The template has them all set at "info" severity, but they're not equally important. If we could set severity at the matcher level, then I could update the template to raise the severity for the more important security headers.

Alternatively, if I could exclude matchers by name, that would solve my issue and would save people from having to argue about matcher severity. I'd imagine my config.yml would look something like this:
```yaml
exclude-matchers:
  - "http-missing-security-headers:cross-origin-resource-policy"
  - "tech-detect:google-font-api"
```

**Comment by olearycrew** [comment]
@ehsandeep is this something we should look to support? Should I open an issue on the Nuceli project for this?

**Comment by princechaddha** [comment]
@carlin-q-scott @olearycrew, We do have support for [excluding matchers](https://github.com/projectdiscovery/nuclei/pull/2218). You can exclude matchers using the `-em` flag or by using `exclude-matchers` in the configuration file, as you mentioned.
